controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -141,6 +141,20 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
 		}
 	}
 
+	// If DataImage exists, add its ownerReference
```

Comment:
nit: Could this be a function we call, instead of dropping more code into the main Reconcile function?

Reply:
Good idea, I will move this to a separate function.